File: Objective-C/TCMPPDemo/CustomApi/TCMPPPaySucessVC.m
//
//  TCMPPPaySucessVC.m
//  TCMPPDemo
//
//  Created by gavinjwxu on 2024/8/12.
//

#import "TCMPPPaySucessVC.h"
#import "UIView+TZLayout.h"
#import "TCMPPCommonTools.h"

@implementation TCMPPPaySucessVC

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.view.backgroundColor = UIColor.whiteColor;
    [self initUI];
}

- (void)initUI {
    UILabel *title = [[UILabel alloc] initWithFrame:CGRectMake(0, 30 + [TCMPPCommonTools safeAreaInsets].top, 200, 20)];
    title.textColor = [UIColor colorWithRed:96.f/255 green:193.f/255 blue:116.f/255 alpha:1];
    title.textAlignment = NSTextAlignmentCenter;
    title.tz_centerX = self.view.tz_centerX;
    title.font = [UIFont boldSystemFontOfSize:18];
    title.text = NSLocalizedString(@"Pay Successfully", nil);
    [self.view addSubview:title];
    
    UIImageView *icon = [[UIImageView alloc] initWithFrame:CGRectMake(0, title.tz_bottom + 100, 60, 60)];
    icon.tz_centerX = title.tz_centerX;
    icon.layer.cornerRadius = 30.f;
    icon.layer.masksToBounds = YES;
    [icon setImage:[UIImage imageNamed:@"tmf_weapp_icon_default"]];
    [TCMPPCommonTools getImageWith:self.iconURL completion:^(UIImage * _Nonnull image, NSError * _Nonnull error) {
        if (image) {
            icon.image = image;
        }
    }];
    [self.view addSubview:icon];
    
    UILabel *name = [[UILabel alloc] initWithFrame:CGRectMake(0, icon.tz_bottom + 20, self.view.tz_width, 20)];
    name.textColor = [UIColor blackColor];
    name.textAlignment = NSTextAlignmentCenter;
    name.tz_centerX = self.view.tz_centerX;
    name.font = [UIFont systemFontOfSize:18];
    name.text = self.name;
    [self.view addSubview:name];
    
    UILabel *price = [[UILabel alloc] initWithFrame:CGRectMake(0, name.tz_bottom + 50, self.view.tz_width, 40)];
    price.textColor = [UIColor blackColor];
    price.textAlignment = NSTextAlignmentCenter;
    price.tz_centerX = self.view.tz_centerX;
    price.font = [UIFont systemFontOfSize:38];
    price.text = [NSString stringWithFormat:@"$ %.2f", self.price/100.f];
    [self.view addSubview:price];
    
    UIButton *doneBtn = [[UIButton alloc] initWithFrame:CGRectMake(0, self.view.tz_bottom - 200, 200, 50)];
    doneBtn.backgroundColor = [UIColor colorWithRed:242.f/255 green:242.f/255 blue:242.f/255 alpha:1];
    doneBtn.titleLabel.textColor = UIColor.blackColor;
    [doneBtn setTitleColor:UIColor.blackColor forState:UIControlStateNormal];
    [doneBtn setTitle:NSLocalizedString(@"Done", nil) forState:UIControlStateNormal];
    [doneBtn addTarget:self action:@selector(doneClick) forControlEvents:UIControlEventTouchUpInside];
    doneBtn.tz_centerX = self.view.tz_centerX;
    doneBtn.layer.cornerRadius = 8.f;
    doneBtn.layer.masksToBounds = YES;
    [self.view addSubview:doneBtn];
}

- (void)doneClick {
    __weak typeof(self) weakSelf = self;
    [self dismissViewControllerAnimated:YES completion:^{
        __strong typeof(self) strongSelf = weakSelf;
        if (strongSelf.dismissBlock) {
            strongSelf.dismissBlock();
        }
    }];
}

@end

File: Swift/TCMPPDemo-Swift/Class/TCMPPPaySucessVC.swift
//
//  TCMPPPaySucessVC.swift
//  TCMPPDemo-Swift
//
//  Created by gavinjwxu on 2024/8/14.
//

import UIKit

class TCMPPPaySucessVC: UIViewController {
    var iconURL: String = ""
    var name: String = ""
    var price: Int = 0
    var dismissBlock: (() -> Void)?

    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .white
        initUI()
    }

    func initUI() {
        let title = UILabel(frame: CGRect(x: 0, y: 30 + TCMPPCommonTools.safeAreaInsets.top, width: 200, height: 20))
        title.textColor = UIColor(red: 96.0 / 255, green: 193.0 / 255, blue: 116.0 / 255, alpha: 1)
        title.textAlignment = .center
        title.center.x = view.center.x
        title.font = UIFont.boldSystemFont(ofSize: 18)
        title.text = NSLocalizedString("Pay Successfully", comment: "")
        view.addSubview(title)

        let icon = UIImageView(frame: CGRect(x: 0, y: title.frame.maxY + 100, width: 60, height: 60))
        icon.center.x = title.center.x
        icon.layer.cornerRadius = 30.0
        icon.layer.masksToBounds = true
        icon.image = UIImage(named: "tmf_weapp_icon_default")

        TCMPPCommonTools.getImageWith(iconURL) { (image, error) in
            if let image = image {
                icon.image = image
            }
        }
        view.addSubview(icon)

        let nameLabel = UILabel(frame: CGRect(x: 0, y: icon.frame.maxY + 20, width: view.frame.width, height: 20))
        nameLabel.textColor = .black
        nameLabel.textAlignment = .center
        nameLabel.center.x = view.center.x
        nameLabel.font = UIFont.systemFont(ofSize: 18)
        nameLabel.text = name
        view.addSubview(nameLabel)

        let priceLabel = UILabel(frame: CGRect(x: 0, y: nameLabel.frame.maxY + 50, width: view.frame.width, height: 40))
        priceLabel.textColor = .black
        priceLabel.textAlignment = .center
        priceLabel.center.x = view.center.x
        priceLabel.font = UIFont.systemFont(ofSize: 38)
        priceLabel.text = String(format: "$ %.2f", Double(price) / 100.0)
        view.addSubview(priceLabel)

        let doneBtn = UIButton(frame: CGRect(x: 0, y: view.frame.maxY - 200, width: 200, height: 50))
        doneBtn.backgroundColor = UIColor(red: 242.0 / 255, green: 242.0 / 255, blue: 242.0 / 255, alpha: 1)
        doneBtn.setTitleColor(.black, for: .normal)
        doneBtn.setTitle(NSLocalizedString("Done", comment: ""), for: .normal)
        doneBtn.addTarget(self, action: #selector(doneClick), for: .touchUpInside)
        doneBtn.center.x = view.center.x
        doneBtn.layer.cornerRadius = 8.0
        doneBtn.layer.masksToBounds = true
        view.addSubview(doneBtn)
    }

    @objc func doneClick() {
        dismiss(animated: true) {
            self.dismissBlock?()
        }
    }
}
